pyscf/tools/trexio.py

```diff
@@ -223,3 +226,88 @@ def _group_by(a, keys):
     assert all(keys[:-1] <= keys[1:])
     idx = np.unique(keys, return_index=True)[1]
     return np.split(a, idx[1:])
+
+def get_occsa_and_occsb(mcscf, norb, nelec, ci_threshold=1e-8):
+    ci_coeff = mcscf.ci
+    num_determinants = int(np.sum(np.abs(ci_coeff) > ci_threshold))
+    occslst = fci.cistring.gen_occslst(range(norb), nelec // 2)
+    selected_occslst = occslst[:num_determinants]
+
+    occsa = []
+    occsb = []
+    ci_values = []
+
+    for i in range(min(len(selected_occslst), mcscf.ci.shape[0])):
+        for j in range(min(len(selected_occslst), mcscf.ci.shape[1])):
+            ci_coeff = mcscf.ci[i, j]
+            if np.abs(ci_coeff) > ci_threshold:  # Check if CI coefficient is significant compared to user defined value
+                occsa.append(selected_occslst[i])
+                occsb.append(selected_occslst[j])
+                ci_values.append(ci_coeff)
+
+    # Sort by the absolute value of the CI coefficients in descending order
+    sorted_indices = np.argsort(-np.abs(ci_values))
+    occsa_sorted = [occsa[idx] for idx in sorted_indices]
+    occsb_sorted = [occsb[idx] for idx in sorted_indices]
+    ci_values_sorted = [ci_values[idx] for idx in sorted_indices]
+
+    return occsa_sorted, occsb_sorted, ci_values_sorted, num_determinants
+
+def det_to_trexio(mcscf, norb, nelec, ci_threshold_or_filename, filename=None, backend='h5', chunk_size=100000):
+    from trexio_tools.group_tools import determinant as trexio_det
+
+    # Determine if ci_threshold is provided
+    if filename is None:
+        filename = ci_threshold_or_filename
+        ci_threshold = 1e-8  # Default value
+    else:
+        ci_threshold = ci_threshold_or_filename
+
+    mo_num = norb
+    int64_num = int((mo_num - 1) / 64) + 1 
+    occsa, occsb, ci_values, num_determinants = get_occsa_and_occsb(mcscf, norb, nelec, ci_threshold)
+
+    det_list = []
+    for a, b, coeff in zip(occsa, occsb, ci_values):
+        occsa_upshifted = [orb + 1 for orb in a]
+        occsb_upshifted = [orb + 1 for orb in b]
+        det_tmp = []
+        det_tmp += trexio_det.to_determinant_list(occsa_upshifted, int64_num)
+        det_tmp += trexio_det.to_determinant_list(occsb_upshifted, int64_num)
+        det_list.append(det_tmp)
+
+    if num_determinants > chunk_size:
+        n_chunks = math.ceil(num_determinants / chunk_size)
+    else:
+        n_chunks = 1 
+
+    with trexio.File(filename, 'w', back_end=_mode(backend)) as tf: 
```

Comment:
you need unsafe `'u'` mode to delete things from TREXIO file (e.g. `determinant` group as here)